knowledge-base/treeview-drag-to-custom-target.md

```diff
@@ -0,0 +1,209 @@
+page_title: How to Drag TreeView Items to a Custom Target
+slug: treeview-kb-drag-to-custom-target
+position: 
+tags: 
+ticketid: 1592132
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>TreeView for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+I have a TreeView that allows dragging and reordering items inside the component. I want to also allow dragging items outside the TreeView, so the users can drop them on custom target. How to achieve that?
+
+## Solution
+
+To allow dragging outside the TreeView and dropping on a custom target, follow these steps:
+
+1. Handle the [`OnDragStart`]({%slug treeview-events%}#drag-events) event of the TreeView to get the item that the user started dragging.
+1. Handle the [`@onpointerup`](https://developer.mozilla.org/en-US/docs/Web/API/Element/pointerup_event) event of your custom target to add the TreeView inside as needed. Based on product specifics, the TreeView internally does not use draggable events, it uses pointer events to handle the dragging. Therefore, `@ondrop` of your custom target will not fire when dragging a TreeView item in it as the component expects a subsequent pointer event. Use the `@onpointerup` instead of `@ondrop`.
+1. Change the icon in the drop clue when the user drags over the custom target to indicate it is an allowed target. Once the [Drag Clue Template](https://feedback.telerik.com/blazor/1501043-drag-clue-template) is available, you may use it to change the rendering as needed. At the dime of writing, changing the icon is only possible with CSS. For that purpose:
+   1. Handle [`@onpointerenter`](https://developer.mozilla.org/en-US/docs/Web/API/Element/pointerenter_event) and [`@onpointerout`](https://developer.mozilla.org/en-US/docs/Web/API/Element/pointerout_event) events of the custom target to detect when the user drags the item over it.
+   1. Raise and drop a flag based on the pointer position.
+   1. Conditionally apply custom CSS to change the drop clue icon based on the flag.
+
+>caption Draggable Treeview that allows dropping items in a custom target
+
+````CSHTML
+<TelerikTreeView Data="@Data"
+                 @bind-ExpandedItems="@ExpandedItems"
+                 Draggable="true"
+                 OnDragStart="@OnDragStart"
+                 OnDrop="@TreeViewDropHandler">
+    <TreeViewBindings>
+        <TreeViewBinding ParentIdField="Parent"></TreeViewBinding>
+    </TreeViewBindings>
+</TelerikTreeView>
+
+@* <TelerikSvgIcon Icon="@SvgIcon.Plus"></TelerikSvgIcon> - Render the desired icon to get its path. *@
+
+<div class="custom-drop-target"
+     @onpointerup="@CustomTargetDropHandler"
+     @onpointerenter="@((PointerEventArgs args) => IsPointerOverCustomTarget = true)"
+     @onpointerout="@((PointerEventArgs args) => IsPointerOverCustomTarget = false)">
+
+     @foreach (var item in DroppedItems)
+    {
+        @item.Text 
+        <br/>
+    }
+</div>
```

Comment:
Consider semantic markup

```suggestion
<ul class="custom-drop-target"
     @onpointerup="@CustomTargetDropHandler"
     @onpointerenter="@((PointerEventArgs args) => IsPointerOverCustomTarget = true)"
     @onpointerout="@((PointerEventArgs args) => IsPointerOverCustomTarget = false)">

     @foreach (var item in DroppedItems)
    {
        <li>@item.Text </li>
    }
</ul>
```

Reply:
This would complicate the drag clue customization due to the nested elements. Let's keep it like this for simplicity.